python/grass/script/imagery.py

```diff
@@ -89,22 +85,37 @@ def group_to_dict(
             .strip()
             .split()
         )
-    except CalledModuleError:
-        return group_dict
+    except CalledModuleError as cme:
+        raise cme
 
-    for idx, raster_map in enumerate(maps_in_group):
-        raster_map_info = raster_info(raster_map, env=env)
-        semantic_label = raster_map_info["semantic_label"]
-        if not raster_map_info["semantic_label"]:
-            warning(
-                _(
-                    "Raster map {rm} in group {igroup} does not have a semantic label."
-                    "Using the numeric row index in the (sub-) group"
-                ).format(rm=raster_map, igroup=imagery_group_name)
+    if subgroup and not maps_in_group:
+        warning(
+            _("Empty result returned for subgroup <{sg}> in group <{g}>").format(
+                sg=subgroup, g=imagery_group_name
             )
-            semantic_label = idx + 1
-        if dict_key == "semantic_label":
-            group_dict[semantic_label] = raster_map_info if full_info else raster_map
-        else:
-            group_dict[raster_map] = raster_map_info if full_info else semantic_label
+        )
+
+    for idx, raster_map in enumerate(maps_in_group):
+        raster_map_info = None
+        if full_info or dict_key == "semantic_labels":
+            raster_map_info = raster_info(raster_map, env=env)
+
+        if dict_key == "indices":
+            key = str(idx + 1)
+            val = raster_map_info or raster_map
+        elif dict_key == "map_names":
+            key = raster_map
+            val = raster_map_info or str(idx + 1)
+        elif dict_key == "semantic_labels":
+            key = raster_map_info["semantic_label"]
+            if not key or key == '"none"':
+                warning(
+                    _(
+                        "Raster map {m} in group <{g}> does not have a semantic label."
+                    ).format(m=raster_map, g=imagery_group_name)
+                )
+                if fill_semantic_label:
+                    key = str(idx + 1)
```

Comment:
This is not what I meant. No warning should be issued when filling in is requested in the parameters.

I'm actually confused with this change because comparing to the previous code, it is filling only when set, what it is doing in the other case? My point was fill it silently when requested to do so. Fail when missing but filling is not requested.

Reply:
Addressed in [85d36a0](https://github.com/OSGeo/grass/pull/3756/commits/85d36a0c83efe0eea2d386b8dd68c970fd326f45)